cpp/src/arrow/compute/api_vector.h

```diff
@@ -257,6 +257,38 @@ class ARROW_EXPORT ListFlattenOptions : public FunctionOptions {
   bool recursive = false;
 };
 
+/// \brief Options for inverse_permutation function
+class ARROW_EXPORT InversePermutationOptions : public FunctionOptions {
+ public:
+  explicit InversePermutationOptions(int64_t max_index = -1,
+                                     std::shared_ptr<DataType> output_type = NULLPTR);
+  static constexpr char const kTypeName[] = "InversePermutationOptions";
+  static InversePermutationOptions Defaults() { return InversePermutationOptions(); }
+
+  /// \brief The max value in the input indices to process. Any indices that are greater
+  /// than this value will be ignored. If negative, this value will be set to the length
+  /// of the input indices minus 1.
+  int64_t max_index = -1;
+  /// \brief The type of the output inverse permutation. If null, the output will be of
+  /// the same type as the input indices, otherwise must be integer types. An invalid
+  /// error will be reported if this type is not able to store the length of the input
+  /// indices.
+  std::shared_ptr<DataType> output_type = NULLPTR;
```

Comment:
Should nullable being considered here?

Reply:
It'll be decided during the actual computation. If there are "holes" in the output, validity buffer will be allocated and filled on demand.